### DISCUSSION TITLE
How to check my device is in sync and how to list problematic files ?

### DISCUSSION THREAD

**Discussion by hilsonp** [vrtmrz/obsidian-livesync#377] [Q&A]
I'm running the obsidian-livesync plugin on 2 windows laptops and one iphone.
I'm currently using the 0.19.10 version. 

Why so ???
Because, my 3 devices are syncing but... as of today, I don't know how to know thay are in sync.

Before upgrading, I want to, on each device:

- verify that all files are in sync
- list problematic files and be able to resolve the conflict

Once all devices are synched and all conflicts are resolved, then, I'll shutdown obsidian, take a backup of a vault on a windows laptop, then start obsidian on the 3 devices and upgrade the plugin on each one.

My questions are:

- How to verify on a device that it is in sync with the couchdb ?
  - On windows, I used "scan storage and database again"
  - On iOS, I cannot find this command. 
- How to list conflicted files ?
  - On windows ?
  - On iOS ?

Until I can safely verify data integrity and resolve conflict on each platform, I'll not be confident...
And I find this a bit scary :-(

You know I support your work which I find awesome.
You may probably tell me that I should just update and that all that is available in the latest release.
I'm just in a deadlock situation because... I don't know the state of my vault scattered over 3 devices :-(

Your insight is welcome. 

**Edit:** A quick update after a manual diff of my 3 vaults 🎉
I just did stop all my obsidian instances, zipped their vault and did a manual diff.
All files are identical (205MB made up of 128 folders and 692 files (mostly md, some images and a few pdf)
except:
- on one device some left-over empty folders (which is a bug you solved a long time ago but I probably did not clean up all my devices)
- one md file conflict not resolved. This is a conflict I did on purpose to try to understand how to deal with it. This can be ignored until I upgrade the plugin to the latest version.
- one pdf file that shows the same content (and pdf file properties) but is 5MB on my windows vault (where it was added initially) but is 32MB on iOS. This only happened with this one file. I'll keep an eye on it but my idea is to move all pdf to an paperless instance and use links to those document instead of storing them in obsidian.

**Comment by vrtmrz** [comment]
Sorry for being very delayed!! I did not notice almost all of the discussions! Somehow I got the notification disabled. 

I had fixed the `Verify all` button on the hatch pane to check the mismatch of the local database and storage, but it was not enough.
So, fixed that feature today again, and we can check the mismatch between the local database and storage and fix that.
After the storage and local database are matched, we can synchronise the local database and the remote database even if they have conflicts.
Then we can perform a `Pick a file to resolve conflict` to check what files are conflicted or all fine.

**Reply by hilsonp** [reply]
Thank you.
